### ISSUE TITLE
wrong test of HAVE_* macros

### ISSUE THREAD

**Issue by DexterMagnific** [libcheck/check#195]
Hi,

When you detect that a function is missing, you define HAVE_FOO=0 in the command line (via add_definitions) and test it in the code like this:

```c
#if defined(FOO)
...
#else
...
#endif
```

This is wrong since FOO is defined, regardless of its value.

The fix consists in either changing `#if defined(HAVE_FOO)` to `#if HAVE_FOO` or to remove the add_definitions() when the valus is zero.

source: https://stackoverflow.com/a/21312684

Regards

**Comment by mikkoi** [comment]
You are right. The difference is subtle but meaningful. Like assert() macro and NDEBUG. Even if NDEBUG=0, assert deactivates itself. These need to be checked.
